bitcoin/_posts/0000-00-00-advent-27.md

```diff
@@ -0,0 +1,287 @@
+
+# 2022: The Year of the Covenant
+
+## BIP-119 Timeline
+CheckTemplateVerify is getting close to ready to go. There are numerous
+[supporters (listed on utxos.org)](https://utxos.org/signals), few detractors,
+and a bumper crop of amazing use cases (did you read the calendar?) waiting for
+us on the other side of CTV activation. The major critiques are that we might want
+something that does 'more' than CTV itself, or to include it in a bundle of
+things.
+
+My take: we're not Gordon Gecko. Greed is _Not_ Good. CTV represents a clean, well
+contained, unproblematic upgrade that's going to deliver hella functionality in
+service of scaling, decentralization, self custody, and privacy. Let's secure
+the bag for Bitcoin users everywhere and make it happen. We can always do more,
+later, informed by what extensions we need for rapidly maturing tools like
+Sapio. CTV is also technically specified and implemented sufficiently -- a view
+I've confirmed with a couple other devs -- that it is able to be considered for
+release.
+
+### What would have to happen to release CTV?
+
+1. More signalers would need to be on
+[utxos.org/signals](https://utxos.org/signals) or other platforms to demonstrate
+interest and demand for CTV, ideally explaining which use cases are important to
+them and why. Every voice counts for consensus.
+1. More "regular contributors" would need to spend time reviewing the code and BIP to
+assure themselves of correctness and safety.  Nothing can move forward with out,
+no matter the count of casual contributors. Many regular contributors don't want
+to 'get political' and look at forks. Fortunately, while all consensus changes
+are complex, CTV is a very tiny and easy to review change in comparison with
+SegWit or Taproot (more similar to CheckLockTimeVerify). **NOTE:** This is a big if!
+Every contributor has the right to review, and ACK or provide a reasoned NACK.
+Even if everyone else is excited about something doesn't mean there isn't space
+for new, thought through, dissent.
+1. We would need to merge the implementation. This is simple, but enough ACKs have to
+   come in and rebases on other subsequent changes to get it in. This can
+   happen ahead of 'full consensus' since there are no deployment parameters,
+   but aids in increasing the testing priority of CTV.
+1. We would need to get a majority of miners/pools primed to accept the upgrade.
+1. We would need to decide on release parameters for the implementation.
+1. We would need to merge and release a client with release parameters
+1. The client would needs to lock-in by a supermajority of miners signalling.
+1. Then, and only then, would CTV be fully available.
+
+### What's the Maximally Aggressive/Optimistic Timeline?
+
+1. Soft Signals / Developer Review: 2-3 months required to get ACKs on the
+implementation, assuming no major changes required.
+1. Merge: Instant, just whenever there are ACKs against the implementation being
```

Comment:
How many ACKs is reasonable?

Reply:
it's not specified anywhere what that would mean for consensus changes. Usually it's 2 ACKs + maintainer for merge, but for consensus it is sort of a quality v.s. quantity + external consensus.